src/Microsoft.ComponentDetection.Orchestrator/Services/ScanExecutionService.cs

```diff
@@ -38,14 +38,17 @@ public async Task<ScanResult> ExecuteScanAsync(ScanSettings settings)
         using var scope = this.logger.BeginScope("Executing BCDE scan");
 
         var detectorRestrictions = this.GetDetectorRestrictions(settings);
-        var detectors = this.detectorRestrictionService.ApplyRestrictions(detectorRestrictions, this.detectors).ToImmutableList();
+        var detectors = this.detectors.ToList();
```

Comment:
Added this line to convert to a list cause the tests were failing with IEnumerable

Reply:
nit: but I think the tests were not mocked properly, we should have to enlist the entries to make then work. Just your check may need to be looking for equivalent objects not the same exact object.

Reply:
The initial one that fails is actually only checking if `DetectorsNotInRun` is not empty ![image](https://github.com/microsoft/component-detection/assets/8052624/11814459-0c9c-4146-a3f0-cf4a12900d77)